app/controllers/products_controller.rb

```diff
@@ -0,0 +1,76 @@
+class ProductsController < ApplicationController
```

Comment:
In these controllers you have no security.  Nothing prevents guest users from getting into mischief.